src/controller/python/chip/yaml/runner.py

```diff
@@ -0,0 +1,387 @@
+        self._test_spec_definition = test_spec_definition
+        self._dev_ctrl = dev_ctrl
+        self._context = _ExecutionContext(data_model_lookup=PreDefinedDataModelLookup())
+
+    def _invoke_action_factory(self, test_step, cluster: str):
+        '''Creates cluster invoke action command from TestStep.
+
+        Args:
+          'test_step': Step containing information required to run an invoke command action.
+          'cluster': Name of cluster action is targeting.
+        Returns:
+          InvokeAction if 'test_step' is a valid action to be executed.
+          None if we were unable to use the provided 'test_step' for a known reason that is not
+          fatal to test execution.
+        '''
+        try:
+            return InvokeAction(test_step, cluster, self._context)
+        except ParsingError:
+            return None
+
+    def _attribute_read_action_factory(self, test_step, cluster: str):
+        '''Creates read attribute command TestStep.
+
+        Args:
+          'test_step': Step containing information required to run read attribute action.
+          'cluster': Name of cluster read attribute action is targeting.
+        Returns:
+          ReadAttributeAction if 'test_step' is a valid read attribute to be executed.
+          None if we were unable to use the provided 'test_step' for a known reason that is not
+          fatal to test execution.
+        '''
+        try:
+            return ReadAttributeAction(test_step, cluster, self._context)
+        except ParsingError:
+            return None
+
+    def _attribute_write_action_factory(self, test_step, cluster: str):
+        '''Creates write attribute command TestStep.
+
+        Args:
+          'test_step': Step containing information required to run write attribute action.
+          'cluster': Name of cluster write attribute action is targeting.
+        Returns:
+          WriteAttributeAction if 'test_step' is a valid write attribute to be executed.
+          None if we were unable to use the provided 'test_step' for a known reason that is not
+          fatal to test execution.
+        '''
+        try:
+            return WriteAttributeAction(test_step, cluster, self._context)
+        except ParsingError:
+            return None
+
+    def encode(self, request) -> BaseAction:
+        action = None
+        cluster = request.cluster.replace(' ', '').replace('/', '')
+        command = request.command
+        # Some of the tests contain 'cluster over-rides' that refer to a different
+        # cluster than that specified in 'config'.
+        if command == 'writeAttribute':
+            action = self._attribute_write_action_factory(request, cluster)
+        elif command == 'readAttribute':
+            action = self._attribute_read_action_factory(request, cluster)
+        elif command == 'readEvent':
+            action = self._event_read_action_factory(request, cluster)
+        else:
```

Comment:
You probably already know this, but there are subscribe ones too.

Reply:
Yeah that is coming next. Hopefully by EOD tomorrow :crossed_fingers: . Getting this PR in makes that a little easier for me